File: documentation/manual/detailledTopics/configuration/code/detailedtopics/configuration/gzipencoding/Global.java
package detailedtopics.configuration.gzipencoding;

//#global
import play.GlobalSettings;
import play.api.mvc.EssentialFilter;
import play.filters.gzip.GzipFilter;

public class Global extends GlobalSettings {
    public <T extends EssentialFilter> Class<T>[] filters() {
        return new Class[]{GzipFilter.class};
    }
}
//#global

File: framework/src/sbt-fork-run-plugin/src/sbt-test/fork-run/dev-mode/app/Global.scala
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
 */

import java.util.Date

import play.api._
import java.io.{FileWriter, File}

object Global extends GlobalSettings {
  override def onStart(app: Application) = {
    // open for append
    val writer = new FileWriter(app.getFile("target/reload.log"), true)
    writer.write(new Date() + " - reloaded\n")
    writer.close()

    if (app.configuration.getBoolean("fail").getOrElse(false)) {
      throw new RuntimeException()
    }
  }
}
